File: tests/Web.AcceptanceTests/Pages/RegisterPage.cs
using System.Text.RegularExpressions;

namespace LiftAndShift.Web.AcceptanceTests.Pages;

public class RegisterPage(IPage page) : BasePage(page)
{
    public override string PagePath => $"{BaseUrl}/register";

    public Task SetEmail(string email) => Page.FillAsync("#email", email);

    public Task SetPassword(string password) => Page.FillAsync("#password", password);

    public Task ClickRegister() => Page.Locator("button[type='submit']").ClickAsync();

    public Task AssertRedirectedToLogin() => Assertions.Expect(Page).ToHaveURLAsync(new Regex(".*/login$"));

    public Task AssertStillOnRegisterPage() => Assertions.Expect(Page).ToHaveURLAsync(new Regex(".*/register$"));

    public Task AssertPasswordHelperTextVisible() =>
        Assertions.Expect(Page.Locator("#password-helper")).ToHaveTextAsync("Password must be at least 6 characters.");
}

File: src/Web/ClientApp/e2e/pages/register.page.ts
import { expect } from '@playwright/test';
import { BasePage, baseUrl } from './base-page';

export class RegisterPage extends BasePage {
  get pagePath(): string {
    return `${baseUrl()}/register`;
  }

  setEmail(email: string) {
    return this.page.fill('#email', email);
  }

  setPassword(password: string) {
    return this.page.fill('#password', password);
  }

  clickRegister() {
    return this.page.locator("button[type='submit']").click();
  }

  assertRedirectedToLogin() {
    return expect(this.page).toHaveURL(/.*\/login$/);
  }

  assertStillOnRegisterPage() {
    return expect(this.page).toHaveURL(/.*\/register$/);
  }

  assertPasswordHelperTextVisible() {
    return expect(this.page.locator('#password-helper')).toHaveText(
      'Password must be at least 6 characters.'
    );
  }
}
